src/struct_info.rs

```diff
@@ -279,6 +286,96 @@ impl<'a> StructInfo<'a> {
                     }
                 }
             }
+            #[doc(hidden)]
+            #[allow(dead_code, non_camel_case_types, non_snake_case)]
+            pub enum #repeated_fields_error_type_name {}
+            #[doc(hidden)]
+            #[allow(dead_code, non_camel_case_types, missing_docs)]
+            impl #impl_generics #builder_name < #( #target_generics ),* > #where_clause {
+                #[deprecated(
+                    note = #repeated_fields_error_message
+                )]
+                pub fn #field_name #generic_arg (self, _: #repeated_fields_error_type_name) -> #builder_name < #( #target_generics ),* > {
+                    self
+                }
+            }
+        })
+    }
+
+    pub fn required_field_impl(&self, field: &FieldInfo) -> Result<TokenStream, Error> {
+        let StructInfo {
+            ref name,
+            ref builder_name,
+            ..
+        } = *self;
+
+        let &FieldInfo {
```

Comment:
Bit of a nitpick, but is there a particular reason why you use `&FieldInfo` here but `*self` when destructuring the `StructIInfo`?